R/obic.R

```diff
@@ -574,66 +600,235 @@ obic_field_dt <- function(dt,output = 'all') {
+#' @param th_obi_p (numeric) A vector with the threshold values for farm evaluation of the physical soil functions, values between 0 and 1, and in increasing order. 
+#' @param th_obi_b (numeric) A vector with the threshold values for farm evaluation of the biological soil functions, values between 0 and 1, and in increasing order. 
+#' @param th_obi_e (numeric) A vector with the threshold values for farm evaluation of the environmental soil functions, values between 0 and 1, and in increasing order. 
+#' @param th_obi_m (numeric) A vector with the threshold values for farm evaluation of the soil management functions, values between 0 and 1, and in increasing order. 
+#' 
+#' @import data.table
+#' 
+#' @examples 
+#'  
+#' \dontrun{
+#' obic_farm(dt = data.table(B_SOILTYPE_AGR = 'rivierklei',B_GWL_CLASS = "II",
+#' B_GWL_GLG = 75,B_GWL_GHG = 10,
+#' B_GWL_ZCRIT = 50,B_SC_WENR = '2',B_HELP_WENR = "MOb72",B_AER_CBS = 'LG01',
+#' B_LU_BRP = c( 1010, 1010,263,263, 263,265,265,265),A_SOM_LOI = 3.91,A_SAND_MI = 66.3,
+#' A_SILT_MI = 22.8,A_CLAY_MI = 7.8,A_PH_CC = 5.4,A_N_RT = 1528.33,A_CN_FR = 13.02,
+#' A_S_RT = 321.26,A_N_PMN = 63.3,A_P_AL = 50.2,A_P_CC = 2.9,A_P_WA = 50.5,
+#' A_CEC_CO = 56.9,A_CA_CO_PO = 66.87,A_MG_CO_PO = 13.97,A_K_CO_PO = 3.06,
+#' A_K_CC = 58.6,A_MG_CC = 77.53,A_MN_CC = 7586.61,A_ZN_CC = 726.2,A_CU_CC = 68.8,
+#' A_C_BCS = 1,A_CC_BCS = 1,A_GS_BCS = 1,A_P_BCS = 1,A_RD_BCS = 1,A_EW_BCS = 1,
+#' A_SS_BCS = 1,A_RT_BCS = 1,A_SC_BCS = 1,M_COMPOST = 0,M_GREEN = FALSE,M_NONBARE =FALSE,
+#' M_EARLYCROP = FALSE,M_SLEEPHOSE = FALSE,M_DRAIN = FALSE,M_DITCH = FALSE,
+#' M_UNDERSEED = FALSE,M_LIME = FALSE,M_MECHWEEDS = FALSE,M_NONINVTILL = FALSE,
+#' M_PESTICIDES_DST = FALSE,M_SOLIDMANURE = FALSE,M_SSPM = FALSE,M_STRAWRESIDUE = FALSE))
+#'}
+#'
+#' @details 
+#' The data.table should contain all required inputs for soil properties needed to calculate OBI score. Management information is optional as well as the observations from the visual soil assessment.
+#' The threshold values per category of soil functions need to have an equal length, with fractions defining the class boundaries in increasing order. 
+#' The lowest boundary value (zero) is not needed.
+#'  
+#' @return 
+#' The output of the Open Bodem Index Calculator for a series of agricultural fields belonging to a single farm. 
+#' Depending on the output type, different output objects can be returned.
+#' These include the estimated OBI scores (both total and aggregated subscores), the value of the underling indicators as well the possible recommendations to improve the soil quality.
+#' The output is a list with field properties as well as aggregated farm properties
+#' 
+#' @export
+obic_farm <- function(dt,
+                      th_obi_c = c(0.4,0.8,1.0),
+                      th_obi_p = c(0.4,0.8,1.0),
+                      th_obi_b = c(0.4,0.8,1.0),
+                      th_obi_e = c(0.4,0.8,1.0),
+                      th_obi_m = c(0.4,0.8,1.0)
```

Comment:
Is there an usecase to dynamically set the thresholds? Because then confusion could arise about which score corresponds to for example `low`?

Reply:
it is practical for our use in the BBWP-OBI. There are no fixed criteria for this selection on farm level. The article of Ros et al. (2022) suggests to use three classes (<0.5, 0.5-0.75 and 0.75-1.0) somehow reflecting the classic soil fertility classes used for soil chemical functions. Its for discussion whether we allow a dynamic setting in the OBI service.

Reply:
after discussion we agreed to remove this dynamic input from the function arguments and to fix this in the function itself. In that way we have more control on the actual calculation / evaluation of soil quality by others using the OBIC, thereby avoiding undesired variation in the OBIC evaluations.